lib/ansible/modules/monitoring/icinga2_host.py

```diff
@@ -0,0 +1,332 @@
+                return True
+        return False
+
+    def create(self, hostname, data):
+        ret = self.call_url(
+            path="v1/objects/hosts/" + hostname,
+            data=self.module.jsonify(data),
+            method="PUT"
+        )
+        return ret
+
+    def delete(self, hostname):
+        data = {"cascade": 1}
+        ret = self.call_url(
+            path="v1/objects/hosts/" + hostname,
+            data=self.module.jsonify(data),
+            method="DELETE"
+        )
+        return ret
+
+    def modify(self, hostname, data):
+        ret = self.call_url(
+            path="v1/objects/hosts/" + hostname,
+            data=self.module.jsonify(data),
+            method="POST"
+        )
+        return ret
+
+    def diff(self, hostname, data):
+        ret = self.call_url(
+            path="v1/objects/hosts/" + hostname,
+            method="GET"
+        )
+        changed = False
+        ic_data = ret['data']['results'][0]
+        for key in data['attrs']:
+            if key not in ic_data['attrs'].keys():
+                changed = True
+            elif data['attrs'][key] != ic_data['attrs'][key]:
+                changed = True
+        return changed
+
+
+# ===========================================
+# Module execution.
+#
+def main():
+    # use the predefined argument spec for url
+    argument_spec = url_argument_spec()
+    # add our own arguments
+    argument_spec.update(
+        state=dict(default="present", choices=["absent", "present"]),
+        name=dict(required=True, aliases=['host']),
+        zone=dict(),
+        template=dict(default=None),
+        check_command=dict(default="hostalive"),
+        display_name=dict(default=None),
+        ip=dict(required=True),
+        variables=dict(type='dict', default=None),
+    )
+
+    # Define the main module
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    state = module.params["state"]
+    name = module.params["name"]
+    zone = module.params["zone"]
+    template = []
+    template.append(name)
+    if module.params["template"]:
+        template.append(module.params["template"])
+    check_command = module.params["check_command"]
+    ip = module.params["ip"]
+    display_name = module.params["display_name"]
+    if not display_name:
+        display_name = name
+    variables = module.params["variables"]
+
+    try:
+        icinga = icinga2_api()
+        icinga.module = module
+        icinga.check_connection()
+    except Exception as e:
+        module.fail_json(msg="unable to connect to Icinga. Exception message: %s" % (e))
+
+    data = {
+        'attrs': {
+            'address': ip,
+            'display_name': display_name,
+            'check_command': check_command,
+            'zone': zone,
+            'vars': {
+                'made_by': "ansible",
+            },
+            'templates': template,
+        }
+    }
+
+    if variables:
```

Comment:
same here. emtpy dict would update variables. Would it make sense using `variables is not None` here?

Reply:
Same as previous; this is correct accoording too PEP8

Reply:
Not pep8, if a user wants to clear variables by providing a emtpy dict, `if variables:` would be false while `if variables is not None:` wouldn't.  I don't know if this is a use case, that is why I am asking.

Reply:
No it's not a usecase; 'variables' is used to add (update) a basic value. So there is no need to make a difference between empty or None.